packages/server/utils/OpenAIServerManager.ts

```diff
@@ -35,7 +36,19 @@ class OpenAIServerManager {
       })
       return (response.data.choices[0]?.text?.trim() as string) ?? null
     } catch (e) {
-      const error = e instanceof Error ? e : new Error('OpenAI failed to getSummary')
+      if (e instanceof Error) {
+        const rateLimitStatusCode = 429
+        if (
+          (e as AxiosError).response?.status === rateLimitStatusCode &&
+          (e as AxiosError).response?.data?.error
+        ) {
+          sendToSentry((e as AxiosError).response?.data?.error)
+        } else {
+          sendToSentry(e)
+        }
+        return null
+      }
+      const error = new Error('OpenAI failed to getSummary')
```

Comment:
+1 It would be nice to have the short branch first with an early return, or an else here.